sdk/rust/.gitignore

```diff
@@ -0,0 +1 @@
+Cargo.lock
```

Comment:
As far as I recall, conventionally, lock files should be committed. Why is this an exception?

Reply:
When crates are published, lock files are ignored. So because these things aren't built into any sort of independent binary, there is no use in maintaining a lock file. And as far as I can tell with the repo, there is nothing that requires that the rust SDK maintain its lock file.

Reply:
Looking at some popular crates, I see they follow this pattern 👍